### PR TITLE
fix/wipe-code-switch

### PR DESCRIPTION
**Pull request by hynek-jina**
Adding option to remove or change wipe code.
Product wise consulted with @Hermez-cz

### CODE REVIEW

File: packages/suite/src/views/settings/SettingsDevice/WipeCode.tsx [hunk 1]
```diff
@@ -20,11 +22,25 @@ interface Props {
...
         dispatch(changeWipeCode({ remove: false }));
+        if (!isDeviceProtectedByWipeCode) {
+            analytics.report({
+                type: EventType.SettingsDeviceSetupWipeCode,
+            });
+        } else {
+            analytics.report({
+                type: EventType.SettingsDeviceChangeWipeCode,
+            });
+        }
```
Comment (komret) [inline packages/suite/src/views/settings/SettingsDevice/WipeCode.tsx]: Simplification:

```suggestion
            analytics.report({
                type: isDeviceProtectedByWipeCode 
                    ? EventType.SettingsDeviceChangeWipeCode
                    : EventType.SettingsDeviceSetupWipeCode,
            });

```

File: packages/suite/src/views/settings/SettingsDevice/WipeCode.tsx [hunk 2]
```diff
@@ -39,11 +55,35 @@ export const WipeCode = ({ isDeviceLocked }: Props) => {
...
+            )}
+            {isDeviceProtectedByWipeCode && (
+                <ActionColumn>
+                    <ActionButton
+                        onClick={enableWipeCode}
+                        isDisabled={isDeviceLocked}
+                        variant="danger"
+                    >
+                        <Translation id="TR_CHANGE_WIPE_CODE" />
+                    </ActionButton>
```
Comment (komret) [inline packages/suite/src/views/settings/SettingsDevice/WipeCode.tsx]: Simplification:

```suggestion
                <ActionColumn>
                    <ActionButton
                        onClick={enableWipeCode}
                        isDisabled={isDeviceLocked}
                        variant="danger"
                    >
                        <Translation id={isDeviceProtectedByWipeCode
                            ? "TR_CHANGE_WIPE_CODE" 
                            : "TR_SETUP_WIPE_CODE"} />
                    </ActionButton>
                </ActionColumn>
```

File: packages/suite/src/actions/settings/deviceSettingsActions.ts [hunk 3]
```diff
@@ -77,7 +77,11 @@ export const changeWipeCode =
         });
         if (result.success) {
             if (!skipSuccessToast) {
```
Comment (komret) [inline packages/suite/src/actions/settings/deviceSettingsActions.ts]: `skipSuccessToast` param is not used, please remove it. We always want to show the toast.

File: packages/suite/src/actions/settings/deviceSettingsActions.ts [hunk 4]
```diff
@@ -77,7 +77,11 @@ export const changeWipeCode =
...
-                dispatch(notificationsActions.addToast({ type: 'wipe-code-changed' }));
+                if (params.remove) {
+                    dispatch(notificationsActions.addToast({ type: 'wipe-code-removed' }));
+                } else {
+                    dispatch(notificationsActions.addToast({ type: 'wipe-code-changed' }));
+                }
```
Comment (komret) [inline packages/suite/src/actions/settings/deviceSettingsActions.ts]: Simplification:
```suggestion
dispatch(
    notificationsActions.addToast({ type: remove ? 'wipe-code-removed' : 'wipe-code-changed' })
);
```

(`params` can be destructured on line 66: `{remove}`)

File: suite-common/wallet-core/src/device/deviceReducer.ts [hunk 5]
```diff
@@ -534,6 +534,11 @@ export const selectIsDeviceProtectedByPin = (state: DeviceRootState) => {
     return !!features?.pin_protection;
 };
 
+export const selectIsDeviceProtectedByWipeCode = (state: DeviceRootState) => {
+    const features = selectDeviceFeatures(state);
+    return !!features?.wipe_code_protection;
+};
+
```
Comment (komret) [inline suite-common/wallet-core/src/device/deviceReducer.ts:541]: I would not create a selector just for this, it is only used once and it is very simple. But that's up to (your) personal preference

**Review by komret — CHANGES_REQUESTED** [review]
No tests 🤔 😄

File: suite-common/wallet-core/src/device/deviceReducer.ts (hunk 5 above)
Comment (hynek-jina) [reply suite-common/wallet-core/src/device/deviceReducer.ts:541]: I left this one there for now.

File: packages/suite/src/actions/settings/deviceSettingsActions.ts [hunk 6]
```diff
@@ -73,12 +73,14 @@ export const changeWipeCode =
             device: {
                 path: device.path,
             },
-            ...params,
+            ...{ remove },
```
Comment (komret) [inline packages/suite/src/actions/settings/deviceSettingsActions.ts]:
```suggestion
            remove,
```

**Review by komret — APPROVED** [review]
🏅

**Pull request merged**